conan/tools/premake/premakedeps.py

```diff
@@ -107,16 +107,19 @@ def _format_flags(flags):
                 return ""
             return ", ".join('"%s"' % p.replace('"', '\\"') for p in flags)
 
-        self.includedirs = _format_paths(dep_cpp_info.includedirs)
         self.libdirs = _format_paths(dep_cpp_info.libdirs)
-        self.bindirs = _format_paths(dep_cpp_info.bindirs)
-        self.libs = _format_flags(dep_cpp_info.libs)
+        with_headers = req and req.headers
+        self.includedirs = _format_paths(dep_cpp_info.includedirs if with_headers else [])
+        with_libs = req and req.libs
+        self.libs = _format_flags(dep_cpp_info.libs if with_libs else [])
+        with_run = req and req.run
+        self.bindirs = _format_paths(dep_cpp_info.bindirs if with_run else [])
         self.system_libs = _format_flags(dep_cpp_info.system_libs)
-        self.defines = _format_flags(dep_cpp_info.defines)
-        self.cxxflags = _format_flags(dep_cpp_info.cxxflags)
-        self.cflags = _format_flags(dep_cpp_info.cflags)
-        self.sharedlinkflags = _format_flags(dep_cpp_info.sharedlinkflags)
-        self.exelinkflags = _format_flags(dep_cpp_info.exelinkflags)
+        self.defines = _format_flags(dep_cpp_info.defines if with_libs and with_headers else [])
```

Comment:
I know CMakeDeps does this, but looks a bit iffy to me that you could have libs=False, headers=True and _not_ get the defines of the dependency? Or am I reading this wrong?

Reply:
``CMakeConfigDeps`` already changed this, using only the ``headers`` to invalidate ``defines``. 
```python
        if not self._require.headers:  # If not depending on headers, paths and
            includedirs = defines = None
```